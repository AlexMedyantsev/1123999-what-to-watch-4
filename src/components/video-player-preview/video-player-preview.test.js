import VideoPlayerPreview from "./video-player-preview.jsx";
import React from "react";
import renderer from "react-test-renderer";

const Video = {
  WIDTH: 300,
  HEIGHT: 100,
  INTERVAL_IN_SEC: 1500,
  IS_MUTED: false,
};

const movie = {
  preview: `https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4`,
  image: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
};

it(`VideoPlayerPreview is rendered correctly`, () => {
  const {preview, image} = movie;

  const tree = renderer.create(
      <VideoPlayerPreview
        source={preview}
        poster={image}
        isMuted={Video.IS_MUTED}
        isPlaying={true}
        width={Video.WIDTH}
        height={Video.HEIGHT}
      />, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
