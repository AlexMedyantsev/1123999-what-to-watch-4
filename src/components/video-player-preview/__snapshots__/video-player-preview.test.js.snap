// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayerPreview is rendered correctly 1`] = `
<video>
  Sorry, this preview is not available in your browser
</video>
`;
